src/conversions/chrono.rs

```diff
@@ -0,0 +1,573 @@
+        let dd = date.day() as u8;
+        let h = time.hour() as u8;
+        let m = time.minute() as u8;
+        let s = time.second() as u8;
+        let ns = time.nanosecond();
+        let (ms, fold) = match ns.checked_sub(1_000_000_000) {
+            Some(ns) => (ns / 1000, true),
+            None => (ns / 1000, false),
+        };
+        let tz = self.offset().fix().to_object(py);
+        let tz = tz.cast_as(py).unwrap();
+        let datetime = PyDateTime::new_with_fold(py, yy, mm, dd, h, m, s, ms, Some(&tz), fold)
+            .expect("Failed to construct datetime");
+        datetime.into()
+    }
+}
+
+impl<Tz: TimeZone> IntoPy<PyObject> for DateTime<Tz> {
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        ToPyObject::to_object(&self, py)
+    }
+}
+
+impl FromPyObject<'_> for DateTime<FixedOffset> {
+    fn extract(ob: &PyAny) -> PyResult<DateTime<FixedOffset>> {
+        let dt = <PyDateTime as PyTryFrom>::try_from(ob)?;
+        let ms = dt.get_fold() as u32 * 1_000_000 + dt.get_microsecond();
+        let h = dt.get_hour().into();
+        let m = dt.get_minute().into();
+        let s = dt.get_second().into();
+        let tz = if let Some(tzinfo) = dt.get_tzinfo() {
+            tzinfo.extract()?
+        } else {
+            return Err(PyTypeError::new_err("Not datetime.timezone.tzinfo"));
+        };
+        let dt = NaiveDateTime::new(
+            NaiveDate::from_ymd(dt.get_year(), dt.get_month().into(), dt.get_day().into()),
+            NaiveTime::from_hms_micro(h, m, s, ms),
+        );
+        Ok(DateTime::from_utc(dt, tz))
+    }
+}
+
+impl FromPyObject<'_> for DateTime<Utc> {
+    fn extract(ob: &PyAny) -> PyResult<DateTime<Utc>> {
+        let dt = <PyDateTime as PyTryFrom>::try_from(ob)?;
+        let ms = dt.get_fold() as u32 * 1_000_000 + dt.get_microsecond();
+        let h = dt.get_hour().into();
+        let m = dt.get_minute().into();
+        let s = dt.get_second().into();
+        let tz = if let Some(tzinfo) = dt.get_tzinfo() {
+            tzinfo.extract()?
+        } else {
+            return Err(PyTypeError::new_err("Not datetime.timezone.utc"));
+        };
+        let dt = NaiveDateTime::new(
+            NaiveDate::from_ymd(dt.get_year(), dt.get_month().into(), dt.get_day().into()),
+            NaiveTime::from_hms_micro(h, m, s, ms),
+        );
+        Ok(DateTime::from_utc(dt, tz))
+    }
+}
+
+impl ToPyObject for FixedOffset {
+    fn to_object(&self, py: Python<'_>) -> PyObject {
+        let dt_module = py.import("datetime").expect("Failed to import datetime");
+        let dt_timezone = dt_module
+            .getattr("timezone")
+            .expect("Failed to getattr timezone");
+        let seconds_offset = self.local_minus_utc();
+        let td =
+            PyDelta::new(py, 0, seconds_offset, 0, true).expect("Failed to contruct timedelta");
+        let offset = dt_timezone
+            .call1((td,))
+            .expect("Failed to call timezone with timedelta");
```

Comment:
I am not sure if there is a better way to do this or we should just create our own wrapper for `PyTimeZone` class like python `datetime.timezone`? That one I am not sure how to get the `#[pyclass]` working. Maybe there is a better way to refer to python `datetime.timezone` without doing this?

Reply:
I know it's not ideal, but I couldn't find a better way to do this either, I'll have a deeper look at this this afternoon and let you know if I find something interesting

Reply:
So, the only way we could do this without executing python code, would be if we could construct a `PyTzInfo` instance ourselves, but the docstring above it says that we can't construct one directly. If I understood everything correctly, we would have to create a new class, which inherits form this abstract baseclass, and then implement the conversion between this class and the non abstract ones in `timezone.utc` and `zoneinfo`, but I think we go back to the problem we discussed in the previous PR? https://github.com/PyO3/pyo3/pull/2604#discussion_r960046250

Reply:
Not sure if @davidhewitt have any suggestions for this.

Reply:
Ah, I didn't know there is a C API for this https://github.com/python/cpython/blob/023c51d9d8e2fd91069eea2bf12e373f1c71e9d2/Modules/_datetimemodule.c#L1182-L1205, I only saw https://github.com/python/cpython/blob/023c51d9d8e2fd91069eea2bf12e373f1c71e9d2/Lib/datetime.py#L2301-L2314 so I thought the implementation was from python side.

Reply:
thanks, I'm now using the `PyTimeZone_FromOffset` and `timezone_utc` instead of running python code